### DISCUSSION TITLE
Is vitedge a good solution for Shopify Hydrogen on Cloudflare Worker?

### DISCUSSION THREAD

**Discussion by miroo93** [frandiox/vitedge#72] [Q&A]
Hi @frandiox 

I am currently experiencing issue deploying new Shopify Hydrogen to Cloudflare Worker and was wondering if implementing vitedge will help solve my issues.

I followed  [their recommendation for deployment](https://shopify.dev/custom-storefronts/hydrogen/deployment#deploy-to-cloudflare-workers) but getting the following error when running `wrangler publish`

> Error: Something went wrong with the request to Cloudflare...
> Uncaught ReferenceError: require is not defined
>   at line 33
>  [API code: 10021]

Here is my `wrangler.toml`
```
name = "<worker-name>"
type = "javascript"
account_id = "<account_id>"
workers_dev = true
route = "<my_service_route>.workers.dev/*"
zone_id = ""

[build]
command = "yarn && yarn build"
[build.upload]
format = "service-worker"

[site]
bucket = "dist/client"
entry-point = "."
```

Sorry if this is not the right place to post this question, but I was hoping I could get some help setting up vitedge for Hydrogen if you think it will help publishing my store on Cloudflare worker. 
I  saw you participated to a [somewhat related discussion](https://github.com/Shopify/hydrogen/discussions/105) few months ago on their github and I was wondering if there was any update and/or vitedge was the current viable solution?

Thank you for your time and help.

**Comment by Roy-Nelson-Tous** [comment]
did you use your real account id?

**Comment by frandiox** [comment]
@miroo93 Hello! I don't think you can mix Vitedge and Hydrogen like that. They are basically alternatives to each other and work somewhat differently.

I think I saw the error you mention some time ago but it was fixed in the latest version of Vite. Can you try to make sure you are running the latest version of both Hydrogen and Vite?
